core/check.py

```diff
@@ -156,4 +154,17 @@ async def spamchannel(ctx: commands.Context) -> bool:
         )
     )
 
+    limit = SpamLimit.get_limit(ctx.guild.id, ctx.channel.id)
+
+    if limit < 0:
+        return True
+
+    if ctx.channel.id not in SPAM_CACHE:
+        SPAM_CACHE[ctx.channel.id] = 0
+
+    if SPAM_CACHE[ctx.channel.id] >= limit:
+        raise exceptions.SpamChannelException()
```

Comment:
This can't work.

This counter would only reset when the bot gets restarted. Which is a big no. We have to hold the messages or at least their timestamps in memory and remove them when they get too old.

I already have this exact thing implemented. However, it is not publicly available code, so I can't point you there. I appreciate your involvement, but maybe it would be easier if I self-assigned the issue #137 and did it instead.

If you want to continue working on this PR, you can, of course. These are the criteria for merging:
- Users can use any spamchannel-guarded command at most five times in five minute window.
- For the first five messages of last five minutes the warning is shown, but the command is allowed.
- For 6th+ message only the warning is shown, the command is disallowed and is not run.
- The message cache is per-channel.
- The message cache is automatically invalidated (e.g. every spamchannel check), so only messages in the last five minutes are being held.

Reply:
This is handled in admin module

```Python
    # Tasks

    @tasks.loop(seconds=60.0)
    async def decrease_spam_cache(self):
        for key in check.SPAM_CACHE.keys():
            if check.SPAM_CACHE[key] > 0:
                check.SPAM_CACHE[key] -= 1
```

Reply:
I'm not sure I like this solution, because it spreads one functionality over multiple not-directly-related files.

Reply:
First 4 points are handled by custom config for every guild.

Message cache is per channel.

The last point is now handled in such way that every minute there's one message removed from cache. I think it's not that bad way 'cause it's slowly decaing so if there's peak in bot spam, it's not reseted in few minutes.

Reply:
IKR, I'm trying to find a solution which I could move this function into check module, but I'm not aware of any pythonic smart way. Can I please ask you for your opinion?

Reply:
Are you sure that `check.SPAM_CACHE` is referring to the same object and not a new instance of it?

Reply:
Pretty sure, it's tested and working. If you see commits, I used print to debug this feature.